src/Suave.Tests/Owin.fs

```diff
@@ -148,5 +159,177 @@ let endToEnd =
+      let noContent = OwinMidFunc(fun next -> OwinAppFunc(fun env ->
+        env.[OwinConstants.responseStatusCode] <- box 204
+        Threading.Tasks.Task.FromResult() :> Threading.Tasks.Task
+        ))
+
+      let composedApp =
+        OwinApp.ofMidFunc noContent
+
+      let asserts (result : HttpResponseMessage) =
+        eq "Http Status Code" HttpStatusCode.NoContent result.StatusCode
+        eq "Reason Phrase set by server" "No Content" result.ReasonPhrase
+
+      runWithConfig composedApp |> reqResp Types.GET "/" "" None None DecompressionMethods.GZip id asserts
+
+    testCase "Composed OWIN security middleware and app" <| fun _ ->
+      let noContent = OwinAppFunc(fun env ->
+        env.[OwinConstants.responseStatusCode] <- box 204
+        Threading.Tasks.Task.FromResult() :> Threading.Tasks.Task
+        )
+      
+      let basicAuthMidFunc = OwinMidFunc(fun next -> OwinAppFunc(fun env ->
+        let requestHeaders : IDictionary<string, string[]> = unbox env.[OwinConstants.requestHeaders]
+        // NOTE: currently fails b/c headers are not using StringComparer.OrdinalIgnoreCase.
+        let authHeader = requestHeaders.["Authorization"].[0].Split([|' '|])
+        let credentials =
+          authHeader.[1]
+          |> Convert.FromBase64String
+          |> Encoding.UTF8.GetString
+        match credentials.Split([|':'|]) with
+        | [|"foo";"bar"|] ->
+          let task = next.Invoke(env)
+          task
+        | _ ->
+          env.[OwinConstants.responseStatusCode] <- box 401
+          Threading.Tasks.Task.FromResult() :> Threading.Tasks.Task
+        ))
+
+      let composedApp =
+        OwinApp.ofAppFunc (basicAuthMidFunc.Invoke(noContent))
+
+      let asserts (result : HttpResponseMessage) =
+        eq "Http Status Code" HttpStatusCode.NoContent result.StatusCode
+        eq "Reason Phrase set by server" "No Content" result.ReasonPhrase
+
+      let sendAuthHeader (req : HttpRequestMessage) =
+        req.Headers.Authorization <- Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String("foo:bar"B))
+        req
+
+      runWithConfig composedApp |> reqResp Types.GET "/" "" None None DecompressionMethods.GZip sendAuthHeader asserts
+
+    testCase "Setting a path in Suave should mount an OWIN app at that path and set requestPathBase" <| fun _ ->
+      let ok (env : OwinEnvironment) =
+        let requestPathBase : string = unbox env.[OwinConstants.requestPathBase]
+        printfn "owin.RequestPathBase is %s" requestPathBase
+        //eq "owin.RequestPathBase" requestPathBase "/owin"
+        let requestPath : string = unbox env.[OwinConstants.requestPath]
+        printfn "owin.RequestPath is %s" requestPath
+        if requestPath <> "/app" then
+          env.[OwinConstants.responseStatusCode] <- box 404
+        else () // 200 OK
+        async.Return ()
+
+      let composedApp =
+        pathRegex "/owin(/.+)*" >>= OwinApp.ofApp ok
```

Comment:
Composing an app with a `path`, `pathRegex`, or `pathScan` should probably work the same way as Katana's `IAppBuilder.Map`.